### DISCUSSION TITLE
Heads up: incorrect when clauses

### DISCUSSION THREAD

**Discussion by ulugbekna** [community/community#49300] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

Hi 👋

VS Code team engineer here 🙂

We're reworking how we parse [when clause contexts](https://code.visualstudio.com/api/references/when-clause-contexts) and noticed that we may break some of the when clauses used in your extension. You can read more in this [issue](https://github.com/microsoft/vscode/issues/175540), which describes upcoming changes: new features, breakages, and migration advice (latest VS Code Insiders now has a linter for when clauses and runs the new parser as default). You can also leave us feedback there.

I also noticed that this VS Code Codespaces extension uses `!=~` operator, which we don't support yet. With the next release of VS Code, it will be possible to write `!(foo =~ /bar/)` to express what `!=~` would. 

Thanks for a great extension, it's very popular! 

Best,
Ulugbek

**Comment by osortega** [comment, marked as answer]
Hello!
Thanks for bringing this up, I think this was already addressed and fixed by @joyceerhl.

@joyceerhl do you mind confirming that we don't require further changes?

**Reply by joyceerhl** [reply]
Yeah this is already fixed in main, thanks @osortega!
